modules/light-clients/07-tendermint/types/client_state_test.go

```diff
@@ -201,6 +202,276 @@ func (suite *TendermintTestSuite) TestInitialize() {
+				suite.Require().NoError(err)
+
+				proof, proofHeight = testingpath.EndpointB.QueryProof(key)
+
+				value = channeltypes.CommitAcknowledgement(ibcmock.MockAcknowledgement.Acknowledgement())
+			},
+			expPass: true,
+		},
+		{
+			name: "successful NextSequenceRecv verification",
+			malleate: func() {
+				// send from chainA to chainB since we are proving chainB incremented the sequence recv
+				packet := channeltypes.NewPacket(ibctesting.MockPacketData, 1, testingpath.EndpointA.ChannelConfig.PortID, testingpath.EndpointA.ChannelID, testingpath.EndpointB.ChannelConfig.PortID, testingpath.EndpointB.ChannelID, clienttypes.NewHeight(0, 100), 0)
+
+				// send packet
+				err := testingpath.EndpointA.SendPacket(packet)
+				suite.Require().NoError(err)
+
+				// next seq recv incremented
+				err = testingpath.EndpointB.RecvPacket(packet)
+				suite.Require().NoError(err)
+
+				key := host.NextSequenceRecvKey(packet.GetSourcePort(), packet.GetSourceChannel())
+				merklePath := commitmenttypes.NewMerklePath(string(key))
+				merklePath, err = commitmenttypes.ApplyPrefix(suite.chainB.GetPrefix(), merklePath)
+				suite.Require().NoError(err)
+
+				path, err = suite.chainB.Codec.Marshal(&merklePath)
+				suite.Require().NoError(err)
+
+				proof, proofHeight = testingpath.EndpointB.QueryProof(key)
+
+				value = sdk.Uint64ToBigEndian(packet.GetSequence() + 1)
+
+			},
+			expPass: true,
+		},
+		{
+			name: "successful custom type verification",
+			malleate: func() {
+				// TODO
+			},
+			expPass: true,
+		},
+		{
+			name: "delay time period has passed",
+			malleate: func() {
+				delayTimePeriod = uint64(time.Second.Nanoseconds())
+			},
+			expPass: true,
+		},
+		{
+			name: "delay time period has not passed",
+			malleate: func() {
+				delayTimePeriod = uint64(time.Hour.Nanoseconds())
+			},
+			expPass: false,
+		},
+		{
+			name: "delay block period has passed",
+			malleate: func() {
+				delayBlockPeriod = 1
+			},
+			expPass: true,
+		},
+		{
+			name: "delay block period has not passed",
+			malleate: func() {
+				delayBlockPeriod = 1000
+			},
+			expPass: false,
+		},
+		{
+			"latest client height < height", func() {
+				proofHeight = testingpath.EndpointA.GetClientState().GetLatestHeight().Increment()
+			}, false,
+		},
+		{
+			"failed to unmarshal merkle path", func() {
+				path = []byte("invalid merkle path")
+			}, false,
+		},
+		{
+			"failed to unmarshal merkle proof", func() {
+				proof = invalidProof
+			}, false,
+		},
+		{
+			"consensus state not found", func() {
+				proofHeight = clienttypes.ZeroHeight()
+			}, false,
+		},
+		{
+			"proof verification failed", func() {
+				// change the value being prooved
+				value = []byte("invalid value")
+			}, false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+			testingpath = ibctesting.NewPath(suite.chainA, suite.chainB)
+			testingpath.SetChannelOrdered()
```

Comment:
Is this required for `NextSequenceRecv` verification?

Reply:
yup! Ordered channels [verify the sequence receive](https://github.com/cosmos/ibc-go/blob/9aea0fe5a2098fa104b461c7fd322e5303484514/modules/core/04-channel/keeper/timeout.go#L111) instead of the packet receipt absence for timeouts

Reply:
Sweet, thanks! :)